src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2070,6 +2071,74 @@ public void setLooseEmptyEnabled(boolean enabled)
         EndpointState.LOOSE_DEF_OF_EMPTY_ENABLED = enabled;
     }
 
+    @Override
+    public Map<String, Map<String, Object>> echoAllNodesWithTiming()
```

Comment:
Thank you for your draft! I am aware that this is a draft and not a proper PR yet but I think it is OK to comment on this already.

My conceptual issue with this is that, imagine you have a cluster of 200 nodes and you execute this JMX method. If you have e.g. a cluster of 5 nodes, even 10, it does not look like a big issue. But if you have 100+, then it is.

You have specified that the timeout is 5 seconds, right? So 200x5 = 1000s ~ 16 min. So _at worst_, we would need to wait 16 minutes until this method returns? What happens if you "ctrl+c" nodetool ping? What happens with the echoes being scheduled for the execution?

Also, it would be nice  to have a way to specify DC I want to ping and also rack in specific DC I want to ping. You can add that as flags to nodetool command.